### ISSUE TITLE
BindingResolutionException with `--coverage` when using pcov

### ISSUE THREAD

**Issue by ChristianPavilonis** [pestphp/pest#451]
No matter what I seem to do I can't seem to get coverage with pcov.

Pcov is enabled
```
php -i | grep pcov
pcov
pcov.directory => /app/app
pcov.exclude => none
pcov.initial.memory => 65336 bytes
pcov.initial.files => 64
```
If I check `ini_get('pcov.enabled');` it outputs 1.

I've tried `php -d pcov.enabled=1 ./vendor/bin/pest --coverage`

Seems related to #253.

```
PHP Fatal error:  Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /app/vendor/laravel/framework/src/Illuminate/Container/Container.php:1093
Stack trace:
#0 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(886): Illuminate\Container\Container->notInstantiable()
#1 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\Container\Container->build()
#2 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\Container\Container->resolve()
#3 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\Foundation\Application->resolve()
#4 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\Container\Container->make()
#5 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(249): Illuminate\Foundation\Application->make()
#6 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(182): Illuminate\Foundation\Bootstrap\HandleExceptions->getExceptionHandler()
#7 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(168): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole()
#8 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleException()
#9 {main}
  thrown in /app/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 1093

   Whoops\Exception\ErrorException

  Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /app/vendor/laravel/framework/src/Illuminate/Container/Container.php:1093
Stack trace:
#0 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(886): Illuminate\Container\Container->notInstantiable()
#1 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\Container\Container->build()
#2 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\Container\Container->resolve()
#3 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\Foundation\Application->resolve()
#4 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\Container\Container->make()
#5 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(249): Illuminate\Foundation\Application->make()
#6 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(182): Illuminate\Foundation\Bootstrap\HandleExceptions->getExceptionHandler()
#7 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(168): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole()
#8 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleException()
#9 {main}
  thrown

  at vendor/laravel/framework/src/Illuminate/Container/Container.php:1093
    1089▕         } else {
    1090▕             $message = "Target [$concrete] is not instantiable.";
    1091▕         }
    1092▕
  ➜ 1093▕         throw new BindingResolutionException($message);
    1094▕     }
    1095▕
    1096▕     /**
    1097▕      * Throw an exception for an unresolvable primitive.

      +1 vendor frames
  2   [internal]:0
      Whoops\Run::handleShutdown()
PHP Fatal error:  Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /app/vendor/laravel/framework/src/Illuminate/Container/Container.php:1093
Stack trace:
#0 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(886): Illuminate\Container\Container->notInstantiable()
#1 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(758): Illuminate\Container\Container->build()
#2 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(851): Illuminate\Container\Container->resolve()
#3 /app/vendor/laravel/framework/src/Illuminate/Container/Container.php(694): Illuminate\Foundation\Application->resolve()
#4 /app/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(836): Illuminate\Container\Container->make()
#5 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(249): Illuminate\Foundation\Application->make()
#6 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(182): Illuminate\Foundation\Bootstrap\HandleExceptions->getExceptionHandler()
#7 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(168): Illuminate\Foundation\Bootstrap\HandleExceptions->renderForConsole()
#8 /app/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(204): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException()
#9 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleShutdown()
#10 {main}
  thrown in /app/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 1093
```

**Comment by silvioiannone** [comment]
Same issue with Xdebug 3 when using `XDEBUG_MODE=coverage`.

**Comment by ysfkaya** [comment]
I'm having the same issue with Xdebug as well. When I run the `php artisan test --coverage` command, it does works.

**Comment by catabozan** [comment]
Has anyone found a solution?

**Comment by owenvoke** [comment]
I still haven't been able to replicate this issue with either PCov (which I use daily) or Xdebug 3. 🤔

**Comment by catabozan** [comment]
I fixed the error in my code by removing a custom sniff I created to use it with phpinsights. It was stored in the app folder. Removing that custom sniff class solved the issue and the code coverage works fine.

PHP: 8.1.2
Laravel: 8.80
Xdebug: 3.1.3
PHPUnit: 9.5.13
Pest: 1.12.1

The custom sniff was extending the `SlevomatCodingStandard\Sniffs\Functions\FunctionLengthSniff` class.

I don't know why that was causing the issue, but deleting the file solved it.

**Comment by SimonJnsson** [comment]
I found that i could reproduce the issue as follows and a workaround for now
In a fresh laravel installation
composer require brianium/paratest (this requires paratest 6.4.3)
pestphp/pest
pestphp/pest-plugin-parallel

i would encounter this error.

What i found is that downgrading brianium/paratest to < 6.4.2 solves the issue.

Hopefully this also works as a workaround for others experiencing the issue until it can be properly resolve.

**TLDR: Downgrade brianium/paratest to 6.4.1**

**Comment by danilopolani** [comment]
> I found that i could reproduce the issue as follows and a workaround for now In a fresh laravel installation composer require brianium/paratest (this requires paratest 6.4.3) pestphp/pest pestphp/pest-plugin-parallel
> 
> i would encounter this error.
> 
> What i found is that downgrading brianium/paratest to < 6.4.2 solves the issue.
> 
> Hopefully this also works as a workaround for others experiencing the issue until it can be properly resolve.
> 
> **TLDR: Downgrade brianium/paratest to 6.4.1**

Paratest is already present in the requirements of `pest-plugin-larallel`, why are you installing it directly? 🤔 By the way I use daily XDebug 3 with `XDEBUG_MODE=develop,coverage` and no issue so far.

**Comment by ChristianPavilonis** [comment]
@SimonJnsson That didn't seem to work for me. I may have to try XDebug.

**Comment by bradsi** [comment]
@ChristianPavilonis I have this error as well. Did you manage to make any progress on it?

**Comment by bradsi** [comment]
I was running into the same error and for me the issue was due to having `processUncoveredFiles="true"` in my `phpunit.xml` file. I've removed this and all seems to be fine now.

Not sure exactly what's going wrong here, but I can now run code coverage with Pest and Pcov so I'm happy!

**Comment by ChristianPavilonis** [comment]
@bradsi, That did it. Thanks!
Any idea as to why that breaks it?

**Comment by bradsi** [comment]
@ChristianPavilonis Awesome, glad I could help!
Looks like it's something to do with the routes files being loaded. There's a further discussion on this [here](https://github.com/laravel/laravel/pull/3725)
